api/app-node/retriever/v1alpha1/rerankretriever_types.go

```diff
@@ -0,0 +1,80 @@
+/*
+Copyright 2024 KubeAGI.
+
+Licensed under the Apache License, Version 2.0 (the "License");
+you may not use this file except in compliance with the License.
+You may obtain a copy of the License at
+
+    http://www.apache.org/licenses/LICENSE-2.0
+
+Unless required by applicable law or agreed to in writing, software
+distributed under the License is distributed on an "AS IS" BASIS,
+WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+See the License for the specific language governing permissions and
+limitations under the License.
+*/
+
+package v1alpha1
+
+import (
+	metav1 "k8s.io/apimachinery/pkg/apis/meta/v1"
+
+	node "github.com/kubeagi/arcadia/api/app-node"
+	"github.com/kubeagi/arcadia/api/base/v1alpha1"
+)
+
+// RerankRetrieverSpec defines the desired state of RerankRetriever
+type RerankRetrieverSpec struct {
+	v1alpha1.CommonSpec   `json:",inline"`
+	CommonRetrieverConfig `json:",inline"`
+	// the endpoint of the rerank
+	// TODO: should change to model or worker
+	Endpoint string `json:"endpoint,omitempty"`
```

Comment:
@Abirdcfly Now we can change this to worker.

Reply:
Before we change string `Endpoint` to `worker`, we may need to bind an svc to the `kubeagi` type of worker to provide services to the outside world, and we need to specify the address of this svc in the annotation of the worker or elsewhere, so that the worker can manage the addition and deletion of this svc, and other downstreams that need this svc, for example, rerank here can just access this svc without relating to the state of the corresponding pod of the worker. 

we can make this change next pull request.

Reply:
or we can have a new crd `Reranker` just like `LLM` `Embeder`